src/include/ref_inline.h

```diff
@@ -0,0 +1,118 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#pragma once
+
+/*
+ * __wt_ref_is_root --
+ *     Return if the page reference is for the root page.
+ */
+static WT_INLINE bool
+__wt_ref_is_root(WT_REF *ref)
+{
+    return (ref->home == NULL);
+}
+
```

Comment:
Add a comment here:
```
/*
 * # The ref state API. #
 *
 * 5 functions* (3 macros) are defined to manipulate the ref state. This is a highly sensitive
 * field and protected via the double underscore keyword. Ideally no one will access it without
 * using these functions. Sadly being C we can't do much better than this.
 *
 * __wt_ref_get_state:
 * Get the state of the ref, wraps a relaxed atomic volatile load. At the time of writing this
 * comment this was done to enable TSAN and to enable burying the field behind the 
 * aforementioned double underscore.
 *
 * WT_REF_SET_STATE:
 * Set the ref state. If HAVE_REF_TRACK is defined track where the set call originated from. The
 * ref state tracking is why we use a macro here, as it utilizes gcc identifiers to track the
 * function and line number from where the macro was called.
 *
 * WT_REF_CAS_STATE:
 * Swap in a new state to the ref, again track where the call originated from.
 *
 * __wt_ref_lock:
 * Spin until the state WT_REF_LOCKED is swapped into the ref state field. Once the call to this
 * function completes the caller has exclusive access to the ref.
 *
 * WT_REF_UNLOCK:
 * Effectively wraps WT_REF_SET_STATE however should only be used when returning the ref to the
 * previous state as returned by __wt_ref_lock.
 */
```

Reply:
Added in [1717dbe](https://github.com/wiredtiger/wiredtiger/pull/10345/commits/1717dbe030dab0ddbb2e7ab981c4e95e6a65cfd0). I've made some small tweaks so please review the new comment